### ISSUE TITLE
Have you considered Maximaleft?

### ISSUE THREAD

**Issue by bradrydzewski** [MoritzMaxeiner/contribution-public-license#2]
> What I wanted was a license that's as close as possible in spirit to the MIT 
license, except requiring any modifications to the software to be contributed 
back to the public under the same license.

I saw your post on license discuss, and I was wondering if you considered the maximaleft license? It is by the same author as this fork, and is based off of the blue oak model license (drafted by multiple experienced lawyers) and addresses some of the feedback you received on the mailing list (patents, language, excuse, requirement to publish source, etc). https://github.com/berneout/maximaleft/blob/master/license.md

Some members of the mailing list believe that OSI should align with the four freedoms, and freedom zero gives you the ability to make private changes without having to publish the source. This license is similar to maximaleft but only enforces copyleft on distribution, which may be less controversial: https://github.com/licensezero/temerity/blob/master/temerity.md

Unfortunately it is unlikely Kyle would submit either of these for OSI approval, however, he may support someone submitting a license on his behalf.  Perhaps this is something to consider if either of the above licenses meet your needs.

**Comment by MoritzMaxeiner** [comment]
Thank you for the information and sorry for the late reply! I don't remember seing a notification for this and had worked on other things for the last couple of months.

> I saw your post on license discuss, and I was wondering if you considered the maximaleft license?

Until I read your post, no I hadn't (because I didn't know about it). I've read and considered it now, though, as well as the blue oak model license; here are my concerns about the former (IANAL, this is not legal advice):

- The copyleft section is the strongest I've seen so far. It would mean that when I use a text editor under this license, I'd have to contribute all source code I write in it. That's too extensive for my use case, where I do want people to able to both use the software and develop other, non-derivative software without forcing hem to contribute. Only derivatives of the already covered software should be touched for my use case.
- Contribution doesn't require keeping at least one publication available for some time frame (publish & instant delete loophole afaiu)
- I'm not currently convinced of the usefulness of the trade mark clause: It can't protect you from non-contributors, so, if need be, a malicious actor can just shift trade mark rights to a different legal entity they control beforehand.
- The excuse period is too short for my comfort (should be more akin to 90 days; even good actors can move slowly at times) and while I think I understand the desire behind the "only once" restriction, I feel this is too restrictive. _In writing_ can include electronic writing, afaik, so the cost of setting up a periodic E-Mail reminder (e.g. every excuse period) until they get the message and comply is not prohibitive.

> Some members of the mailing list believe that OSI should align with the four freedoms, and freedom zero gives you the ability to make private changes without having to publish the source. 

I do not agree with freedom 0 allowing this. Here is freedom 0 as defined [by the FSF](https://www.gnu.org/philosophy/free-sw.html):

> The freedom to run the program as you wish, for any purpose (freedom 0).

Being allowed to run the program however you wish does not automatically include being allowed to create private modifications without publishing them.

Freedom 1 comes closer, but it doesn't require you to be able to keep your changes private:
> The freedom to study how the program works, and change it so it does your computing as you wish (freedom 1). Access to the source code is a precondition for this. 

Freedom 2 doesn't touch private modifications at all:
> The freedom to redistribute copies so you can help others (freedom 2).

Freedom 3 requires allowing to distribute modified copies, it does not exclude prohibiting not distributing modified copies:
> The freedom to distribute copies of your modified versions to others (freedom 3). By doing this you can give the whole community a chance to benefit from your changes. Access to the source code is a precondition for this. 

So, to the best of my understanding, requiring all modifications to be public is not contrary to free software, as it is currently defined by the FSF. But I get that FSF people feel otherwise (in which case they should adjust their freedom definitions to reflect this clearly, at least imho).

Regardless of that, Free software (FSF) and Open Source software (OSI) are, by their respective current definitions, not the same and I consider pretending otherwise to be a recipe for desaster down the line. If the actual goal is to change those definitions to be the same, then my remark would be "whats the point of having both, then? Just merge already!".

> This license is similar to maximaleft but only enforces copyleft on distribution, which may be less controversial:

This, unfortunately doesn't cover my use case, because creating derivatives and getting around contribution by not distributing them is the exact issue I have with most existing licenses. There's nothing stopping a company, for example, from taking a GPL-2 (or similarly licensed) OS, make significant performance improvements to it, run it on their own servers (with massive benefits to them), and never contributing a single line of code back.

> Unfortunately it is unlikely Kyle would submit either of these for OSI approval, however, he may support someone submitting a license on his behalf. Perhaps this is something to consider if either of the above licenses meet your needs.

Sadly, for the reasons listed above, neither of these are suitable for my needs.
